sdk/network/mgmt/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationSecurityGroupImpl.java

```diff
@@ -45,6 +45,6 @@ public String resourceGuid() {
 
     @Override
     public String provisioningState() {
-        return inner().provisioningState();
```

Comment:
Should we just give the `ProvisioningState` object?

Reply:
I prefer to return string. It looks a few interfaces just return the object. Maybe we can do in another PR to align such behavior.

Reply:
I think object would be better for comparison. Since it is an enum.